src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuotes.tsx

```diff
@@ -94,6 +96,20 @@ export const TradeQuotes: React.FC<TradeQuotesProps> = memo(
       [activeQuoteIndex, bestQuoteData?.quote?.steps, isLoading, sortedQuotes],
     )
 
+    // add some loading state per swapper so missing quotes have obvious explanation
+    const fetchingSwappers = useMemo(() => {
+      return Object.entries(isSwapperFetching)
+        .filter(([_swapperName, isFetching]) => isFetching)
+        .map(([swapperName, _isFetching]) => {
+          return (
+            <FetchingTradeQuote
+              key={`${swapperName}-fetching`}
```

Comment:
could use `i` instead for paranoia - `swapperName` may not necessarily be unique (even though it accidentally is now)